### PR TITLE
Update ODBC documentation

### PR DESCRIPTION
**Pull request by jordanw-bq**
*Issue #, if available:*

*Description of changes:*
-clearing up instructions that are unclear
-updating out-of-date commands/file paths

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: sql-odbc/docs/dev/BUILD_INSTRUCTIONS.md [hunk 1]
```diff
@@ -2,85 +2,100 @@
...
+* Driver library: `./build/odbc/lib/libodfesqlodbc.dylib`
+* Test binaries folder: `./build/odbc/bin/`
 
 ### Packaging
 
-Run below command from the project's build directory.
->cpack .
+```
+cd cmake-build64/
+cpack .
```
Comment (dai-chen) [inline sql-odbc/docs/dev/BUILD_INSTRUCTIONS.md:92]: What I found in ODBC GitHub action is the following 4 commands:

```
$ cd cmake-build64
$ cmake ../src
$ make
$ cpack .
```

Is the cmake and make command also required? Or it's already covered in other section.

Comment (jordanw-bq) [reply sql-odbc/docs/dev/BUILD_INSTRUCTIONS.md:92]: The cmake & make commands seem to be there to make sure the CMake steps have been run for packaging with GHActions. For active development, these lines shouldn't be needed, but I'll add a note around this block to reflect that these are needed if driver/CMake changes are made

Comment (jordanw-bq) [reply sql-odbc/docs/dev/BUILD_INSTRUCTIONS.md:96]: Added note to re-run build script before packaging, if driver code/CMake project changes have been made

Comment (dai-chen) [reply sql-odbc/docs/dev/BUILD_INSTRUCTIONS.md:96]: I see. Thanks!

**Review by dai-chen — APPROVED** [review]
Thanks for the update!

**Review by penghuo — APPROVED** [review]
Thanks for the change.

**Pull request merged**